src/stcal/tweakreg/tweakreg.py

```diff
@@ -253,14 +254,43 @@ def _parse_refcat(abs_refcat: str | Path,
         )
 
     if Path(abs_refcat).is_file():
-        return Table.read(abs_refcat)
+        return _parse_sky_centroid(Table.read(abs_refcat))
 
     msg = (f"Invalid 'abs_refcat' value: {abs_refcat}. 'abs_refcat' must be "
            "a path to an existing file name or one of the supported "
            f"reference catalogs: {_SINGLE_GROUP_REFCAT_STR}.")
     raise ValueError(msg)
 
 
+def _parse_sky_centroid(catalog: Table) -> Table:
+    """Turn SkyCoord object into simple RA/DEC columns.
+
+    The inclusion of SkyCoord objects via sky_centroid.ra and sky_centroid.dec
+    permits the use of catalogs directly from the jwst source_catalog step.
+    No action is taken if the catalog already contains RA and DEC columns.
+    """
+    cols = [name.lower() for name in catalog.colnames]
```

Comment:
Can a table contain `RA`, `rA`, `Ra`, and `ra` columns at the same time? If "Yes", then I think this code should be enhanced and throw an exception if there are multiple instances of `ra` in `cols` list. While it may never happen, we must catch something like this.

Reply:
Or, maybe implement a logic like: if `RA` and `DEC` are present - use those, if not try `ra` and `dec`, if not use `lower()` and try again...

Reply:
good suggestion, this is changed in the most recent version.  I think it's best for the step to fail if it sees e.g. `RA` and `ra`, so that the user doesn't accidentally get unexpected results from the wrong column.